src/content_delegate.js

```diff
@@ -585,31 +588,193 @@ ContentDelegate.prototype.attachRecapLinkToEligibleDocs = function() {
+        let pacer_doc_id = $(this.links[i]).data('pacer_doc_id');
+        if (!pacer_doc_id) {
+          continue;
+        }
+        let result = api_results.results.filter(function (obj) {
+          return obj.pacer_doc_id === pacer_doc_id;
+        })[0];
+        if (!result) {
+          continue;
+        }
+        let href = `https://www.courtlistener.com/${result.filepath_local}`;
+        let recap_link = $('<a/>', {
+          'class': 'recap-inline',
+          'title': 'Available for free from the RECAP Archive.',
+          'href': href
+        });
+        recap_link.click($.proxy(this.handleRecapLinkClick, this, window, href));
+        recap_link.append($('<img/>').attr({
+          src: chrome.extension.getURL('assets/images/icon-16.png')
+        }));
+        recap_link.insertAfter(this.links[i]);
       }
-      let href = `https://www.courtlistener.com/${result.filepath_local}`;
-      let recap_link = $('<a/>', {
-        'class': 'recap-inline',
-        'title': 'Available for free from the RECAP Archive.',
-        'href': href
-      });
-      recap_link.click($.proxy(this.handleRecapLinkClick, this, window, href));
-      recap_link.append($('<img/>').attr({
-        src: chrome.extension.getURL('assets/images/icon-16.png')
-      }));
-      recap_link.insertAfter(this.links[i]);
+    }, this));
+};
+
+// TODO: implement appellate court check
+// TODO: Confirm that zip downloading is consistent across jurisdictions
+ContentDelegate.prototype.onDownloadAllSubmit = async function (event) {
+  // helper function - extract the zip by creating an html Element and then querying for the frame
+  const extractUrl = (html) => {
+    const page = document.createElement("html");
+    page.innerHTML = html;
+    const frames = page.querySelectorAll("iframe");
+    return frames[0].src;
+  };
+
+  // helper function - convert string to html document
+  const stringToDocBody = (str) => {
+    const parser = new DOMParser();
+    const newDoc = parser.parseFromString(str, 'text/html');
+    return newDoc.body;
+  };
+
+  // helper function - returns filename based on user preferences
+  const generateFileName = (options, pacerCaseId) => {
+    if (options.ia_style_filenames) {
+      return [
+        'gov',
+        'uscourts',
+        this.court,
+        (pacerCaseId || 'unknown-case-id')
+      ].join('.').concat('.zip');
+    } else if (options.lawyer_style_filenames) {
+      const firstTable = document.getElementsByTagName('table')[0];
+      const firstTableRows = firstTable.querySelectorAll('tr');
+      // 4th from bottom
+      const matchedRow = firstTableRows[firstTableRows.length - 4];
+      const cells = matchedRow.querySelectorAll('td');
+      const document_number = cells[0].innerText.match(/\d+(?=\-)/)[0];
+      const docket_number = cells[1].innerText;
+      return [
+        PACER.COURT_ABBREVS[this.court],
+        docket_number,
+        document_number,
+      ].join('_').concat('.zip');
```

Comment:
There's a lot of fragile stuff here. Is it worth putting in a try with a simple file name if something goes wrong (like you do for the ia_style_filename branch?

Reply:
Yeah I’m agnostic about this. I copied this from the showPDFpage function. Feel free to take a crack as I won’t be back at my desk until later.